### DISCUSSION TITLE
Adding Variable Speed Propeller Functionality in Unreal

### DISCUSSION THREAD

**Discussion by rotebrink** [JSBSim-Team/jsbsim#975] [Q&A]
I come from an interesting position, I'm actually more of an artist and I feel like I'm getting into things that are very technical here. My day job is 3D modelling and texturing for a small game studio, and I have very basic knowledge about using Unreal. Basically I'm playing around with setting up a new personal project, and I've been a fan of flight sims since I was a kid.

To the question, I would like to know how I would go about adding variable speed propeller functionality in Unreal Engine. Based on what I've read so far, and I'm really not a programmer (only some entry level Python experience from a while back, for fun), my best guess is that I need to add some lines to the jsbsimmovementcomponent. Using the Aeromatic tools from the JSBSim website, I created the xml files for a basic WW2 fighter. If I make a prop file with fixed pitch, it works (although I'm slow, which may have to do with the set pitch?), if I make it variable speed, no dice. I'm also guessing that has to do with the fact that variable speed propellers probably aren't set up in the UE example project file. If I can avoid having to spend years learning C++, that would be preferred, only because I would rather stick to what I'm already good at.

Thanks so much for any help, feeling really stupid here, he he.

**Comment by gallonmate** [comment]
Define "no dice" and what exactly are you trying to emulate. There are a few sample aircraft included with the plugin that have variable pitch propellers, like the P51D for example. Propeller pitch can already be set in the Unreal Engine plugin:

If you look at the default project and open the default BP_airliner blueprint, go to the Tick Section of the graph. In the first line of that tick code, second section is where the engine commands are set. You can click the node that says Set members in Engine Commands and then enable the pin to access the Propeller Advance which controls the propeller pitch. The input value is normalize from 0 to 1 with 0 as min pitch and 1 as max pitch. The nodes in circled in Yellow is just a quick example I added for the screenshot.
![image](https://github.com/JSBSim-Team/jsbsim/assets/28702753/30e87bf4-856d-41c8-83cd-19afe5a6463d)

**Reply by rotebrink** [reply]
Wonderful, thank you. Will try as soon as I have a chance. I did try flying the 51 as well and was having the same problem, which is in part why I thought it had to do with the variable pitch. You're awesome!

**Reply by gallonmate** [reply]
Glad to help! But I should clarify the P51D can fly without doing anything to the propeller. So that's why I was asking what problem you were having.

**Reply by rotebrink** [reply]
Everything on and full throttle and it wasn't moving for me.
